TS2717: Subsequent property declarations must have the same type.

I get the below error and unable to proceed 

> [ng] ERROR in node_modules/@capacitor/core/dist/esm/core-plugin-definitions.d.ts(8,9): error TS2717: Subsequent property declarations must have the same type.  Property 'Camera' must be of type 'CameraPlugin', but here has type 'CameraPlugin'.
> [ng] node_modules/@capacitor/core/dist/esm/core-plugin-definitions.d.ts(10,9): error TS2717: Subsequent property declarations must have the same type.  Property 'Device' must be of type 'DevicePlugin', but here has type 'DevicePlugin'.
> [ng] node_modules/@capacitor/core/dist/esm/core-plugin-definitions.d.ts(11,9): error TS2717: Subsequent property declarations must have the same type.  Property 'Filesystem' must be of type 'FilesystemPlugin', but here has type 'FilesystemPlugin'.
> [ng] node_modules/@capacitor/core/dist/esm/core-plugin-definitions.d.ts(13,9): error TS2717: Subsequent property declarations must have the same type.  Property 'Haptics' must be of type 'HapticsPlugin', but here has type 'HapticsPlugin'.
> [ng] node_modules/@capacitor/core/dist/esm/core-plugin-definitions.d.ts(16,9): error TS2717: Subsequent property declarations must have the same type.  Property 'Modals' must be of type 'ModalsPlugin', but here has type 'ModalsPlugin'.
> [ng] node_modules/@capacitor/core/dist/esm/core-plugin-definitions.d.ts(19,9): error TS2717: Subsequent property declarations must have the same type.  Property 'Photos' must be of type 'PhotosPlugin', but here has type 'PhotosPlugin'.
> [ng] node_modules/@capacitor/core/dist/esm/core-plugin-definitions.d.ts(23,9): error TS2717: Subsequent property declarations must have the same type.  Property 'StatusBar' must be of type 'StatusBarPlugin', but here has type 'StatusBarPlugin'.

Looking at node_modules/@capacitor/core/dist/esm/core-plugin-definitions.d.ts there are other modules for which no ERROR is thrown like GeolocationPlugin, ClipboardPlugin ...

> $ grep -inr "CameraPlugin" * --include=*.ts #Plugin having trouble
> node_modules/@capacitor/core/dist/esm/core-plugin-definitions.d.ts:8:        Camera?: CameraPlugin;
> node_modules/@capacitor/core/dist/esm/core-plugin-definitions.d.ts:224:export interface CameraPlugin extends Plugin {
> node_modules/@capacitor/core/dist/esm/web/camera.d.ts:2:import { CameraPlugin, CameraPhoto, CameraOptions } from '../core-plugin-definitions';
> node_modules/@capacitor/core/dist/esm/web/camera.d.ts:3:export declare class CameraPluginWeb extends WebPlugin implements CameraPlugin {
> node_modules/@capacitor/core/dist/esm/web/camera.d.ts:8:declare const Camera: CameraPluginWeb;
> $ grep -inr "ClipboardPlugin" * --include=*.ts #Plugin for which no error is shown
> node_modules/@capacitor/core/dist/esm/core-plugin-definitions.d.ts:9:        Clipboard?: ClipboardPlugin;
> node_modules/@capacitor/core/dist/esm/core-plugin-definitions.d.ts:315:export interface ClipboardPlugin extends Plugin {
> node_modules/@capacitor/core/dist/esm/web/clipboard.d.ts:2:import { ClipboardPlugin, ClipboardWrite, ClipboardRead, ClipboardReadResult } from '../core-plugin-definitions';
> node_modules/@capacitor/core/dist/esm/web/clipboard.d.ts:3:export declare class ClipboardPluginWeb extends WebPlugin implements ClipboardPlugin {
> node_modules/@capacitor/core/dist/esm/web/clipboard.d.ts:8:declare const Clipboard: ClipboardPluginWeb;

package.json below
```
{
  "name": "teacher",
  "license": "SEE LICENSE IN <your-license-filename>",
  "version": "0.0.0",
  "scripts": {
    "ng": "ng",
    "start": "ionic serve",
    "build:web": "ng build",
    "build:ionic": "ng build && npm run cap.copy",
    "build:mobile": "npm run cap.copy && npm run cap.update",
    "build:mobile:ios": "npm run cap.copy && npx cap update ios",
    "build:mobile:android": "npm run cap.copy && npx cap update android",
    "build:prod": "npm run build:web -- --prod && npm run build:mobile",
    "cap.add.ios": "npx cap add ios",
    "cap.add.android": "npx cap add android",
    "cap.ios": "npx cap open ios",
    "cap.android": "npx cap open android",
    "cap.copy": "npx cap copy",
    "cap.update": "npx cap update",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e"
  },
  "dependencies": {
    "@angular/animations": "file:../../node_modules/@angular/animations",
    "@angular/common": "file:../../node_modules/@angular/common",
    "@angular/compiler": "file:../../node_modules/@angular/compiler",
    "@angular/core": "file:../../node_modules/@angular/core",
    "@angular/forms": "file:../../node_modules/@angular/forms",
    "@angular/platform-browser": "file:../../node_modules/@angular/platform-browser",
    "@angular/platform-browser-dynamic": "file:../../node_modules/@angular/platform-browser-dynamic",
    "@angular/router": "file:../../node_modules/@angular/router",
    "@app/store": "file:./src/app/store",
    "@capacitor/android": "^1.0.0-beta.6",
    "@capacitor/cli": "^1.0.0-beta.11",
    "@capacitor/core": "^1.0.0-beta.11",
    "@capacitor/ios": "^1.0.0-beta.3",
    "@ionic-native/core": "file:../../node_modules/@ionic-native/core",
    "@ionic-native/splash-screen": "file:../../node_modules/@ionic-native/splash-screen",
    "@ionic-native/status-bar": "file:../../node_modules/@ionic-native/status-bar",
    "@ionic/angular": "file:../../node_modules/@ionic/angular",
    "@ionic/ng-toolkit": "file:../../node_modules/@ionic/ng-toolkit",
    "@ionic/schematics-angular": "file:../../node_modules/@ionic/schematics-angular",
    "@ngrx/effects": "file:../../node_modules/@ngrx/effects",
    "@ngrx/router-store": "file:../../node_modules/@ngrx/router-store",
    "@ngrx/store": "file:../../node_modules/@ngrx/store",
    "@ngx-translate/core": "file:../../node_modules/@ngx-translate/core",
    "@ngx-translate/http-loader": "file:../../node_modules/@ngx-translate/http-loader",
    "@nrwl/nx": "file:../../node_modules/@nrwl/nx",
    "@portal/ionic": "file:../../xplat/ionic",
    "@portal/scss": "file:../../libs/scss",
    "@portal/web": "file:../../xplat/web",
    "rxjs": "file:../../node_modules/rxjs",
    "rxjs-compat": "file:../../node_modules/rxjs-compat",
    "zone.js": "file:../../node_modules/zone.js"
  },
  "devDependencies": {
    "@angular-devkit/architect": "0.7.2",
    "@angular-devkit/build-angular": "0.7.2",
    "@angular-devkit/core": "0.7.2",
    "@angular-devkit/schematics": "0.7.2",
    "@angular/cli": "^6.1.4",
    "@angular/compiler-cli": "file:../../node_modules/@angular/compiler-cli",
    "@angular/language-service": "file:../../node_modules/@angular/language-service",
    "@types/jasmine": "file:../../node_modules/@types/jasmine",
    "@types/jasminewd2": "file:../../node_modules/@types/jasminewd2",
    "@types/node": "file:../../node_modules/@types/node",
    "codelyzer": "file:../../node_modules/codelyzer",
    "jasmine-core": "file:../../node_modules/jasmine-core",
    "jasmine-spec-reporter": "file:../../node_modules/jasmine-spec-reporter",
    "karma": "file:../../node_modules/karma",
    "karma-chrome-launcher": "file:../../node_modules/karma-chrome-launcher",
    "karma-coverage-istanbul-reporter": "file:../../node_modules/karma-coverage-istanbul-reporter",
    "karma-jasmine": "file:../../node_modules/karma-jasmine",
    "karma-jasmine-html-reporter": "file:../../node_modules/karma-jasmine-html-reporter",
    "protractor": "file:../../node_modules/protractor",
    "ts-node": "file:../../node_modules/ts-node",
    "tslint": "file:../../node_modules/tslint",
    "typescript": "file:../../node_modules/typescript"
  }
}

```

Please suggest what could be wrong

--- a/core/src/core-plugin-definitions.ts
+++ b/core/src/core-plugin-definitions.ts
@@ -1,30 +1,32 @@
 import { Plugin, PluginListenerHandle } from './definitions';
 
-declare global {
-  interface PluginRegistry {
-    Accessibility?: AccessibilityPlugin;
-    App?: AppPlugin;
-    BackgroundTask?: BackgroundTaskPlugin;
-    Browser?: BrowserPlugin;
-    Camera?: CameraPlugin;
-    Clipboard?: ClipboardPlugin;
-    Device?: DevicePlugin;
-    Filesystem?: FilesystemPlugin;
-    Geolocation?: GeolocationPlugin;
-    Haptics?: HapticsPlugin;
-    Keyboard?: KeyboardPlugin;
-    LocalNotifications?: LocalNotificationsPlugin;
-    Modals?: ModalsPlugin;
-    Motion?: MotionPlugin;
-    Network?: NetworkPlugin;
-    Photos?: PhotosPlugin;
-    PushNotifications?: PushNotificationsPlugin;
-    Share?: SharePlugin;
-    SplashScreen?: SplashScreenPlugin;
-    StatusBar?: StatusBarPlugin;
-    Storage?: StoragePlugin;
-    Toast?: ToastPlugin;
-    WebView?: WebViewPlugin;
+export interface PluginRegistry {
+  Accessibility?: AccessibilityPlugin;
+  App?: AppPlugin;
+  BackgroundTask?: BackgroundTaskPlugin;
+  Browser?: BrowserPlugin;
+  Camera?: CameraPlugin;
+  Clipboard?: ClipboardPlugin;
+  Device?: DevicePlugin;
+  Filesystem?: FilesystemPlugin;
+  Geolocation?: GeolocationPlugin;
+  Haptics?: HapticsPlugin;
+  Keyboard?: KeyboardPlugin;
+  LocalNotifications?: LocalNotificationsPlugin;
+  Modals?: ModalsPlugin;
+  Motion?: MotionPlugin;
+  Network?: NetworkPlugin;
+  Photos?: PhotosPlugin;
+  PushNotifications?: PushNotificationsPlugin;
+  Share?: SharePlugin;
+  SplashScreen?: SplashScreenPlugin;
+  StatusBar?: StatusBarPlugin;
+  Storage?: StoragePlugin;
+  Toast?: ToastPlugin;
+  WebView?: WebViewPlugin;
+
+  [pluginName: string]: {
+    [prop: string]: any;
   }
 }
 

--- a/core/src/definitions.ts
+++ b/core/src/definitions.ts
@@ -1,6 +1,4 @@
-declare global {
-  export interface PluginRegistry {}
-}
+import { PluginRegistry } from './core-plugin-definitions';
 
 export interface Plugin {
   addListener(eventName: string, listenerFunc: Function): PluginListenerHandle;
@@ -101,3 +99,5 @@ export interface WindowCapacitor {
   };
   window: any;
 }
+
+export { PluginRegistry };

--- a/plugin-template/src/definitions.ts
+++ b/plugin-template/src/definitions.ts
@@ -1,4 +1,4 @@
-declare global {
+declare module "@capacitor/core" {
   interface PluginRegistry {
     Echo?: EchoPlugin;
   }

--- a/site/docs-md/plugins/index.md
+++ b/site/docs-md/plugins/index.md
@@ -44,13 +44,14 @@ The rest of the fields are optional and will fill out your initial `package.json
 ## TypeScript Interface
 
 Each plugin comes with some typescript files that simply export TypeScript interfaces. These interfaces
-can provide typing to TypeScript consumers of your plugin. 
+can provide typing to TypeScript consumers of your plugin.
 
 Starting with the TypeScript interface can be a good way to build out the API for your plugin. For example,
 here's the default interface for our Plugin:
 
 ```typescript
-declare global {
+module
+declare module "@capacitor/core" {
   interface PluginRegistry {
     Echo?: EchoPlugin;
   }
@@ -69,7 +70,7 @@ Whenever you are ready to publish your plugin, just use:
 npm publish
 ```
 
-This will build the JS portion of your plugin and publish the rest of your plugin files to NPM. 
+This will build the JS portion of your plugin and publish the rest of your plugin files to NPM.
 
 Your package can now be installed using `npm install your-plugin` in any Capacitor app.
 

--- a/site/docs-md/plugins/index.md
+++ b/site/docs-md/plugins/index.md
@@ -50,7 +50,6 @@ Starting with the TypeScript interface can be a good way to build out the API fo
 here's the default interface for our Plugin:
 
 ```typescript
-module
 declare module "@capacitor/core" {
   interface PluginRegistry {
     Echo?: EchoPlugin;

